src/dimension.h

```diff
@@ -19,10 +19,19 @@ typedef struct PartitioningInfo PartitioningInfo;
 typedef struct DimensionSlice DimensionSlice;
 typedef struct DimensionVec DimensionVec;
 
+/* Valid values for FormData_dimension fd.type */
+#define FD_DIMENSION_OPEN 'o'
+#define FD_DIMENSION_CLOSED 'c'
+#define FD_DIMENSION_CORRELATED 'C'
+#define FD_DIMENSION_CHECK 'h' /* these are not stored in dimension catalog table */
+#define FD_DIMENSION_ANY 'a'
+
+/* Corresponding enum values  */
 typedef enum DimensionType
 {
 	DIMENSION_TYPE_OPEN,
 	DIMENSION_TYPE_CLOSED,
+	DIMENSION_TYPE_CORRELATED,
 	DIMENSION_TYPE_ANY,
 } DimensionType;
```

Comment:
Can we unify the enum with the char values so that we don't need two sets of values and translate between? You should be able to define the enum values:

```suggestion
typedef enum DimensionType
{
	DIMENSION_TYPE_OPEN = 'o',
	DIMENSION_TYPE_CLOSED = 'c',
	DIMENSION_TYPE_CORRELATED = 'C',
	DIMENSION_TYPE_ANY = 'a',
} DimensionType;
```

Reply:
Done